src/components/security_manager/test/security_manager_test.cc

```diff
@@ -521,109 +536,27 @@ TEST_F(SecurityManagerTest, StartHandshake_SSLInternalError) {
-
-  // Listener was destroyed after OnHandshakeDone call
   mock_sm_listener.release();
 }
-/*
- * Shall send data on call StartHandshake
- */
-TEST_F(SecurityManagerTest, StartHandshake_SSLInitIsNotComplete) {
-  SetMockCryptoManager();
-  uint32_t connection_id = 0;
-  uint8_t session_id = 0;
-  // uint8_t protocol_version = 0;
-  EXPECT_CALL(mock_session_observer, PairFromKey(key, _, _));
-  EXPECT_CALL(mock_session_observer, GetHandshakeContext(key))
-      .Times(3)
-      .WillRepeatedly(Return(SSLContext::HandshakeContext()));
-  EXPECT_CALL(mock_session_observer,
-              ProtocolVersionUsed(connection_id, session_id, _))
-      .WillOnce(Return(true));
 
-  // Expect send one message (with correct pointer and size data)
-  EXPECT_CALL(mock_protocol_handler, SendMessageToMobileApp(_, is_final));
-
-  // Return mock SSLContext
-  EXPECT_CALL(mock_session_observer, GetSSLContext(key, kControl))
-      .Times(3)
-      .WillRepeatedly(Return(&mock_ssl_context_exists));
-  // Expect initialization check on each call StartHandshake
-  EXPECT_CALL(mock_ssl_context_exists, IsInitCompleted())
-      .Times(3)
-      .WillRepeatedly(Return(false));
-  EXPECT_CALL(mock_ssl_context_exists, SetHandshakeContext(_)).Times(3);
-
-  // Emulate SSLContext::StartHandshake with different parameters
-  // Only on both correct - data and size shall be send message to mobile app
-  EXPECT_CALL(mock_ssl_context_exists, StartHandshake(_, _))
-      .WillOnce(DoAll(SetArgPointee<0>(handshake_data_out_pointer),
-                      SetArgPointee<1>(0),
-                      Return(SSLContext::Handshake_Result_Success)))
-      .WillOnce(DoAll(SetArgPointee<0>((uint8_t*)NULL),
-                      SetArgPointee<1>(handshake_data_out_size),
-                      Return(SSLContext::Handshake_Result_Success)))
-      .WillOnce(DoAll(SetArgPointee<0>(handshake_data_out_pointer),
-                      SetArgPointee<1>(handshake_data_out_size),
-                      Return(SSLContext::Handshake_Result_Success)));
-
-  security_manager_->StartHandshake(key);
-  security_manager_->StartHandshake(key);
-  security_manager_->StartHandshake(key);
-}
-/*
- * Shall notify listeners on call StartHandshake after SSLContext initialization
- * complete
- */
-TEST_F(SecurityManagerTest, StartHandshake_SSLInitIsComplete) {
-  SetMockCryptoManager();
-  // Expect no message send
-  // Expect notifying listeners (success)
-  EXPECT_CALL(*mock_sm_listener,
-              OnHandshakeDone(key, SSLContext::Handshake_Result_Success))
-      .WillOnce(Return(true));
-
-  // Emulate SessionObserver result
-  EXPECT_CALL(mock_session_observer, GetSSLContext(key, kControl))
-      .WillOnce(Return(&mock_ssl_context_exists));
-  EXPECT_CALL(mock_ssl_context_exists, IsInitCompleted())
-      .WillOnce(Return(true));
-
-  security_manager_->StartHandshake(key);
-
-  // Listener was destroyed after OnHandshakeDone call
-  mock_sm_listener.release();
-}
```

Comment:
@jacobkeeler these unit tests became not actual after changing implementation details in security_manager. After these changes we can't check actual SSLContext status in the same way, because now handshake could be postponed due to certificate/system time reason.